libraries/abstractions/secure_sockets/test/iot_test_tcp.c

```diff
@@ -2489,17 +2502,17 @@ static void prvServerDomainName( void )
     xResult = prvSetSockOptHelper( xSocket, xReceiveTimeOut, xSendTimeOut );
     TEST_ASSERT_EQUAL_INT32_MESSAGE( SOCKETS_ERROR_NONE, xResult, "Set sockopt Failed" );
 
+    /* Get the address struct for AWS Broker and SetSockOpt REQUIRE_TLS. */
+    xResult = prvAwsIotBrokerConnectHelper( xSocket, &xAwsBrokerAddress );
+    TEST_ASSERT_EQUAL_INT32_MESSAGE( SOCKETS_ERROR_NONE, xResult, "Failed setup AWS Broker address struct." );
+
```

Comment:
What is the purpose of moving this here? (from its previous location)

Reply:
This test case expects that the SNI of an invalid hostname will fail, but the hostname was getting overwritten in the helper function. By entering an invalid hostname after using the helper function, the test's condition holds, and the Socket is unable to connect